### PR TITLE
create helm chart

### PR DESCRIPTION
**Pull request by JeroenRijks**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EM-1262

### Change description ###
create helm charts

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
[x] We'll see
```

### CODE REVIEW

File: charts/em-npa/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name: em-npa
+home: https://github.com/hmcts/rpa-native-pdf-annotator-app
+version: 0.0.1 # WHAT SHOULD THIS BE
```
Comment (timja) [inline charts/em-npa/Chart.yaml]:
```suggestion
version: 0.0.1
```

File: charts/em-npa/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+description: Helm chart for the HMCTS Native PDF Annotator API
```
Comment (timja) [inline charts/em-npa/Chart.yaml]:
```suggestion
description: HMCTS Native PDF Annotator API
```

File: charts/em-npa/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+maintainers:
+  - name: HMCTS EM Team
+    email: pawel.kaczmarek@hmcts.net
```
Comment (timja) [inline charts/em-npa/Chart.yaml]: either team email address or omit the email field ideally

File: charts/em-npa/values.saat.template.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+npa-app:
+  environment:
+    IDAM_API_BASE_URI: https://preprod-idamapi.reform.hmcts.net:3511
```
Comment (timja) [inline charts/em-npa/values.saat.template.yaml]: wrong value for saat, but do you really need the saat file?
I would just move everything to values.template.yaml

File: charts/em-npa/values.template.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+postgresql:
+  initdbScripts:
+    init.sql: |-
+      CREATE USER emnpa WITH PASSWORD 'emnpa';
+      CREATE DATABASE emnpa WITH OWNER = emnpa ENCODING = 'UTF-8' CONNECTION LIMIT = -1;
```
Comment (timja) [inline charts/em-npa/values.template.yaml]: if you only have one DB you can just use the regular config:
https://github.com/hmcts/draft-store/blob/master/charts/draft-store-service/values.template.yaml#L18

File: charts/em-npa/values.template.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    ENDPOINTS_HEALTH_SENSITIVE: false
+    ENDPOINTS_INFO_SENSITIVE: false
+    S2S_NAMES_WHITELIST: em_api,em_gw,ccd_gw,ccd_data,sscs,divorce_document_upload,divorce_document_generator,probate_backend,jui_webapp,pui_webapp
+    CASE_WORKER_ROLES: caseworker-probate,caseworker-cmc,caseworker-sscs,caseworker-divorce
+    ENABLE_IDAM_HEALTH_CHECK: false
+    ENABLE_S2S_HEALTH_CHECK: false
+    ENABLE_LIQUIBASE: true
+
+    # Needed?
+    # CLAIM_STORE_URL: http://${SERVICE_NAME}-alias_defined_in_requirements.yaml
```
Comment (timja) [inline charts/em-npa/values.template.yaml]: ?

File: kubernetes/deployment.template.yaml [hunk 7]
```diff
@@ -111,10 +111,9 @@ spec:
               memory: "2048Mi"
               cpu: "2500m"
           ports:
-            - containerPort: 8080 #TODO Check this is correct.
+            - containerPort: 8080
```
Comment (timja) [inline kubernetes/deployment.template.yaml]: bin this file when you're ready

Comment (JeroenRijks) [reply kubernetes/deployment.template.yaml]: will do

File: charts/em-npa/values.template.yaml (hunk 6 above)
Comment (JeroenRijks) [reply charts/em-npa/values.template.yaml]: copied my setup from CMC, was going to delete this after my tech lead reviews which env vars we need

File: charts/em-npa/values.template.yaml [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  image: ${IMAGE_NAME}
+  ingressHost: ${SERVICE_FQDN}
+  applicationPort: 8080
+  keyVaults:
+    em-npa:
+      resourceGroup: em-npa
+      # Pawel, what secrets do we need in here, and what should they be called?
+      secrets:
+#        - APPINSIGHTS_INSTRUMENTATIONKEY
+        # FUNCTIONAL_TEST_CLIENT_S2S_TOKEN or microservicekey-em-gw?
```
Comment (adusumillipraveen) [inline charts/em-npa/values.template.yaml]: Values in here should be key vault secret key names, not env vars. Env vars can be mapped in bootstrap.yaml

File: src/main/resources/bootstrap.yaml [hunk 9]
```diff
@@ -0,0 +1,13 @@
+spring:
+  cloud:
+    propertiesvolume:
+#      properties are loaded from given paths
+      enabled: true
+#      parent directory prefixes the property names
+      prefixed: true
+      paths: /mnt/secrets/em-npa
+      aliases:
+        em-npa.FUNCTIONAL_TEST_CLIENT_S2S_TOKEN: FUNCTIONAL_TEST_CLIENT_S2S_TOKEN
```
Comment (adusumillipraveen) [inline src/main/resources/bootstrap.yaml]: mapping should be from secrets to env vars. em-npa.{secretkey} : {envvar} is the syntax

**Pull request closed without merging**